### DISCUSSION TITLE
Hyper-V Issue on CloudStack VM Running Windows Server 2019

### DISCUSSION THREAD

**Discussion by Narasimha201997** [apache/cloudstack#9689] [Q&A]
I recently established a Windows Server 2019 instance on CloudStack and activated Hyper-V within the 2019 server virtual machine. Subsequently, the virtual machine entered repair mode. After conducting diagnostics, the virtual machine became operational; however, Hyper-V remains non-functional. When attempting to launch the virtual machine on Hyper-V, the following error is encountered: "Failed to start the Virtual Machine because some of the Hyper-V components are not running." 

I have attempted similar configurations on both Windows Server 2019 and 2022, where Hyper-V operates without any issues. 
I kindly request your assistance in troubleshooting the problem on the Windows Server 2019 instance.

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by btzq** [comment]
Hi @Narasimha201997 , 

Want to confirm the description:

- You have a server running HyperV as Hypervisor right now? 

or 

- You are using a KVM Hypervisor, which is hosting a Guest VM (running on Windows Server 2019), and trying to run HyperV inside that Guest VM, to host another Guest VM?

**Comment by yadvr** [comment]
Is hardware assisted virtualisation enabled on your host & have you configured KVM agent to allow host passthrough?

**Comment by DaanHoogland** [comment]
closed this for lack of interaction, please reopen or open a new item if applicable
